### ISSUE TITLE
Ignore error based on message

### ISSUE THREAD

**Issue by Luc45** [squizlabs/PHP_CodeSniffer#3570]
I'd like to enforce comments on every class method with `Squiz.Commenting.FunctionComment.Missing`, except if the function is a `__construct`.

I'm not seeing a way to accomplish, as the I can either enable/disable this rule globally in the application or for some specific files, but not for the `__construct` function specifically.

To accomplish this, I wanted to ignore an error based on the message I receive, similar to how PHPStan does it:

```
<rule ref="Squiz.Commenting.FunctionComment.Missing">
    <exclude-message>#^Missing doc comment for function __construct\(\)$#</exclude-message>
</rule>
```

Is it currently possible to achieve this in PHPCS? If not, this issue is a feature suggestion to add this functionality, as it gives a lot of flexibility to the rules.

**Comment by jrfnl** [comment]
This is not currently possible with PHPCS.

Generally speaking, either allowing a sniff to be customized via properties or a sniff having (very) modular error code, which would allow to exclude a specific error code is the way to do this in PHPCS.

**Comment by tijmenmoddit** [comment]
Yeah same! need this! or only be able to exclude certain functions like __construct for the rule
